lib/get-config.js

```diff
@@ -303,7 +310,60 @@ function validateRules(config, options) {
   }
 }
 
-module.exports = function(options) {
+function _determineConfigForSeverity(config) {
+  switch (config) {
+    case SEVERITY_LEVEL_IGNORE:
+      return { config: false, severity: IGNORE_SEVERITY };
+    case SEVERITY_LEVEL_WARNING:
+      return { config: true, severity: WARNING_SEVERITY };
+    case SEVERITY_LEVEL_ERROR:
+      return { config: true, severity: ERROR_SEVERITY };
+  }
+}
+
+/**
+ * Sets the severity and config on the rule object.
+ * Eg:
+ * {'no-implicit-this': 'warn'} -> {'no-implicit-this': { severity: 'warn', config: true }}
+ * { 'no-implicit-this': [ 'warn', { allow: [ 'fooData' ] } ] }
+ * would become:
+ * { 'no-implicit-this': { severity: 'warn', config: 'allow': [ 'fooData' ] } }
+ * @param {*} configData
+ */
+function processRules(config) {
+  for (let key in config.rules) {
+    let severityConfig;
+    let ruleData = config.rules[key];
+    let ruleConfig = {};
+    // In case of {'no-implicit-this': 'off|warn|error'}
+    if (typeof ruleData === 'string') {
+      severityConfig = _determineConfigForSeverity(ruleData);
+      if (severityConfig) {
+        ruleConfig.severity = severityConfig.severity;
+        ruleConfig.config = severityConfig.config;
+        config.rules[key] = ruleConfig;
```

Comment:
This should be set outside of the conditional (not in each branch).

Reply:
this was done for the same reason, I wanted to suppress errors caused due to changing the default format while working on another part, need to fix this one